assets/js/components/event/ControlPanel.tsx

```diff
@@ -193,17 +193,27 @@ const ControlPanel = ({
   const { isOpen, onOpen, onClose } = useDisclosure();
   const rerender = useRerender();
 
-  const updateSources = async () => {
+  const updateAvailableSources = async () => {
     const sources = await getSources();
-    if (sources != null) setSources(sources);
+    if (sources != null) {
+      setSources(sources);
+
+      const audio = getCurrentDeviceName(client, "audio");
+      if (audio === undefined)
+        setSourceById(client, sources.audio[0].deviceId, "audio", playerCallback);
+
+      const video = getCurrentDeviceName(client, "video");
+      if (video === undefined)
+        setSourceById(client, sources.video[0].deviceId, "video", playerCallback);
```

Comment:
I would also check if change like this works.
```suggestion
     const prepareDevice = (kind: "audio" | "video") => {
        const deviceName = getCurrentDeviceName(client, kind);
        if (deviceName === undefined)
          setSourceById(client, sources[kind][0].deviceId, kind, playerCallback);
      }
      prepareDevice("audio");
      prepareDevice("video")
```